### PR TITLE
added disabled styling to Ghost and Primary buttons (fixes #62)

### LINKED ISSUE #62: Add disabled prop to button Ghost and Primary
when the button receive a disabled prop the click needs be blocked, need a visual representation too, like a gray color and not-allowed cursor. you can be creative and propose other things

### PR DESCRIPTION
**Pull request by losito105**
Changes made to `Ghost` :
When passed the disabled prop, hovering over the button:
-  shows `not-allowed` cursor
-  changes the opacity of the button to `.5`
-  blocks clicks

Video demo of change:

https://user-images.githubusercontent.com/43625213/156953421-fac44d09-b1df-4e8b-b400-3866bbd6dab4.mov

Changes made to `Primary` :
When passed the disabled prop, hovering over the button:
-  shows `not-allowed` cursor
-  changes the opacity of the button to `.5`
-  blocks clicks

Video demo of change:

https://user-images.githubusercontent.com/43625213/156953443-ee0f98b2-7ff7-47c0-9366-f2eda8703b35.mov

### CODE REVIEW

File: components/button/Primary.tsx [hunk 1]
```diff
@@ -6,23 +6,48 @@ type PrimaryProps = {
...
-        display: block;
-        margin: 0 auto;
-        font-weight: var(--font-bold);
-      }
-    `}</style>
-    {children}
-  </button>
-)
+const Primary = ({ children, ...props }: PrimaryProps) => {
+  if (props.disabled) {
```
Comment (vczb) [inline components/button/Primary.tsx:10]: are not necessary render two distincts buttons. you can add a disabled prop when this exists. i will send you an example based in your code:

```js
import { ButtonHTMLAttributes, ReactNode } from 'react'

type ButtonTypes = ButtonHTMLAttributes<HTMLButtonElement>

type PrimaryProps = {
  children?: ReactNode | string | null
} & ButtonTypes

const Primary = ({ children, ...props }: PrimaryProps) => {
  return (
    <button {...props}  disabled={props?.disabled || false}>
      <style jsx>{`
        button {
          border: none;
          cursor: pointer;
          background: var(--primary);
          color: var(--white);
          width: 100%;
          height: 4rem;
          display: block;
          margin: 0 auto;
          font-weight: var(--font-bold);
        }

        button:disabled {
          border: none;
          cursor: not-allowed;
          background: var(--primary);
          color: var(--white);
          width: 100%;
          height: 4rem;
          display: block;
          margin: 0 auto;
          font-weight: var(--font-bold);
        }
        button:disabled:hover {
          opacity: 0.5;
        }
      `}</style>
      {children}
    </button>
  )
}

export default Primary
```

**Review by vczb — CHANGES_REQUESTED** [review]
Very beautiful styles but the code can be improved. 

if you have any question i'm happy to help

**Comment by losito105** [comment]
@vczb Thank you for the suggestion! That was a definite improvement to what I wrote. Everything should be fixed in my most recent commit.

File: components/button/Ghost.tsx [hunk 2]
```diff
@@ -6,19 +6,32 @@ type GhostProps = {
...
-  </button>
-)
+const Ghost = ({ children, ...props }: GhostProps) => {
+  return (
+    <button {...props} disabled={props?.disabled || false}>
+      <style jsx>{`
+        button {
+          border: none;
+          cursor: pointer;
+          background: transparent;
```
Comment (vczb) [inline components/button/Ghost.tsx:16]: sorry for not have reporting this before, but here have a lot of repeatable rules like background, width, etc. the values are the same. will be good to performance and best practices if on disabled we replace only the rules of we need, for example cursor, opacity...

File: components/button/Primary.tsx [hunk 3]
```diff
@@ -6,23 +6,40 @@ type PrimaryProps = {
...
+          height: 4rem;
+          display: block;
+          margin: 0 auto;
+          font-weight: var(--font-bold);
+        }
+
+        button:disabled {
+          border: none;
+          cursor: not-allowed;
+          background: var(--primary);
```
Comment (vczb) [inline components/button/Primary.tsx:28]: here is the same of ghost, i will make a video to example it

**Review by vczb — CHANGES_REQUESTED** [review]
it's a good git practice we submit only one commit for PR. i will make a video to show you how i can combine one commits or more

https://www.loom.com/share/9780a3004c854ed58ff0e2419741664e

- some rules are the same
- we have more than 1 commit

to combine commits you can use a `interactive rebase` or reset it and add as `git commit --amend --no-edit` 

this git flow it's a little complex, if you need i'm available

**Comment by vczb** [comment]
hello @losito105 waint forward for your changes :) if you have any question i will be happy to help, cheers

**Comment by losito105** [comment]
> hello @losito105 waint forward for your changes :) if you have any question i will be happy to help, cheers

Hey @vczb! It's been a busy week at work so far - I'll get to making these corrections ASAP (most likely on Thursday). Cheers!

**Comment by vczb** [comment]
> > hello @losito105 waint forward for your changes :) if you have any question i will be happy to help, cheers
> 
> Hey @vczb! It's been a busy week at work so far - I'll get to making these corrections ASAP (most likely on Thursday). Cheers!

dont worry its ok  :) 

have a nice week

**Comment by mr3saru** [comment]
@losito105 i changed the base of compare because we want to make or git workflow more easily. 

now you can compare directly with main and we will merge it at some time as approved

**Pull request closed without merging**